submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,185 @@
+const KEYS_VALUE_SEPARATOR = ": ";
+const PROPS_SEPARATOR = "; ";
+
+class Creature {
+    constructor({ name, gender, friends = [], species, saying }) {
+        this.name = name;
+        this.species = species;
+        this.gender = gender;
+        this.friends = friends;
+        this.saying = saying;
+    }
+
+    createArrayOfProps() {
+        return ["name", "species", "gender", "saying", "friends"]
```

Comment:
This is definitely a way to list properties explicitly.
Do we still need to stick to it?
You used yet another way to list properties explicitly, i.e. `{ name, gender, friends = [], species, saying }` in method declaration. It is even better as it lists properties rather than just their names.

The following example also lists properties: `[ propA, propB, propC ]`

Destructuring laso works well on `this`, e.g. `const {propA, propB] = this` that make code DRY while yet explicit.

BTW, the following example lists props and also transforms one of them so its data type becomes similar to other props type: `[ stringA, stringB, arrayA.join(", "), stringC ]`

See how you can leverage the above to improve your code.